lesson_02/load_sales_raw_job1/main.py

```diff
@@ -0,0 +1,200 @@
+sales_dal: SalesDalInterface = SalesDalCourseFake(RAW_SALES_API_URL, RAW_LOAD_API_AUTH_TOKEN)
+sales_bll: SalesApiBlInterface = SalesApiBllCourseFake(
+    sales_dal, storage_dal,
+    RAW_LOAD_CONTINUE_ON_PAGE_FAIL, RAW_SALES_API_REQUESTS_DELAY_SEC, RAW_SALES_API_MAX_FAILS,
+    RAW_FILE_NAME_TEMPLATE)
+
+# -- Create global container end ---
+print('Ready to accept requests')
+
+
+# -- Controller methods implementation begin ---
+@app.route('/', methods=['GET'])
+def root_index():
+    return render_template('index.html')
+
+
+@app.route('/', methods=['POST'])
+def controller_load_sales_raw() -> flask_typing.ResponseReturnValue:
+    """
+    Controller that accepts sales raw data and save locally
+
+    Proposed POST body in JSON:
+    {
+      "date: "2022-12-09",
+      "raw_dir": "/raw/sales/2022-12-09"
+    }
+    """
+
+    log: LogItemInterface = log_item_factory.new_for_request()
+    try:
+        try:
+            input_data: dict = request.json
+        except BaseException as e:
+            log.user_error(e, "Client pass wrong JSON format")
+            return {
+                "message": "Wrong JSON",
+            }, HTTPStatus.BAD_REQUEST
+
+        not_supported_parameters: set = set(input_data.keys()).difference(['date', 'raw_dir'])
+        if len(not_supported_parameters) > 0:
+            return {
+                "message": f'Parameters not supported: {not_supported_parameters}',
+            }, HTTPStatus.IM_A_TEAPOT
+
+        date_str = input_data.get('date')
+        raw_dir = input_data.get('raw_dir')
+
+        if not date_str:
+            log.user_error('"date" parameter missed by client')
+            return {
+                "message": f'"date" parameter missed',
+            }, HTTPStatus.BAD_REQUEST
+        if not raw_dir:
+            log.user_error('"raw_dir" parameter missed by client')
+            return {
+                "message": f'"raw_dir" parameter missed',
+            }, HTTPStatus.BAD_REQUEST
+
+        sales_bll.save_sales_to_storage(log, date_str=date_str, raw_dir=raw_dir)
+
+        return {
+            "message": "Data retrieved successfully from API",
+        }, HTTPStatus.CREATED
+
+    except SalesBllDateFormatException as e:
+        log.user_error(e, "Client pass date with wrong format")
+        return {
+            "message": str(e),
+        }, HTTPStatus.BAD_REQUEST
+    except SalesBllPathFormatException as e:
+        log.user_error(e, "Client pass not acceptable logical file path")
+        return {
+            "message": str(e),
+        }, HTTPStatus.BAD_REQUEST
+    except SalesBllNothingLoadedException as e:
+        log.user_warning(e, "Sales not found")
+        return {
+            "message": f"Sales not found",
+        }, HTTPStatus.NOT_FOUND
+    except SalesBllPartiallyLoadedException as e:
+        log.dev_error(e, f"Sales loaded partially. {e}")
+        return {
+            "message": f"Sales loaded partially. {e}",
+        }, HTTPStatus.PARTIAL_CONTENT
```

Comment:
You try to handle as many exceptions as possible. But in the real world, it usually doesn't work and makes things worse, unfortunately.